google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/VerifyAdvertiserIdentity.java

```diff
@@ -0,0 +1,158 @@
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      System.exit(1);
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      System.exit(1);
+    }
+
+    try {
+      new VerifyAdvertiserIdentity().runExample(googleAdsClient, params.customerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the API client to use.
+   * @param customerId the customer ID to update.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+    try (IdentityVerificationServiceClient identityVerificationServiceClient =
+        googleAdsClient.getLatestVersion().createIdentityVerificationServiceClient()) {
+      // Retrieves the current Advertiser Identity Verification status.
+      IdentityVerification identityVerification =
+          getIdentityVerification(customerId, identityVerificationServiceClient);
+
+      if (identityVerification != null) {
+        if (identityVerification.hasVerificationProgress()
+            && identityVerification.getVerificationProgress().getActionUrl().isEmpty()) {
+          // Starts an identity verification session.
+          startIdentityVerification(customerId, identityVerificationServiceClient);
+
+          // Calls getIdentityVerification again to retrieve the verification progress after
+          // starting an identity verification session.
+          getIdentityVerification(customerId, identityVerificationServiceClient);
+        } else {
+          // If there is an identity verification session in progress, there is no need to start
+          // another one by calling startIdentityVerification.
+          System.out.println("There is an advertiser identify verification session in progress.");
+          System.out.printf(
+              "The URL for the verification process is '%s' and it will expire at '%s'%n",
+              identityVerification.getVerificationProgress().getActionUrl(),
+              identityVerification.getVerificationProgress().getInvitationLinkExpirationTime());
+        }
+      } else {
```

Comment:
Optional style comment: I think you can restructure this to (assuming nothing changes based on my other comment)
```
if (identityVerification == null)
...
else if (...actionUrl is empty)
...
else
...
```

This creates three distinct states without a nested if statement, but your call.

Reply:
Addressed with new approach of `switch` on program status